dali/test/python/test_RN50_external_source_parallel_data.py

```diff
@@ -95,6 +90,9 @@ def iteration_test(args):
                 pipe.reset()
 
         print("OK {}".format(pipe_factory.__name__))
+        # pipline is not immediately released because traceback of last exception kept by
+        # python (i.e. StopIteration) stores reference to the pipeline
+        gc.collect()
```

Comment:
So what this gc does here?  Do we have any guarantee it does something?

Reply:
Okay, turns out that the traceback outliving the scope can be avoided and was due to circular reference to traceback created when raising the exception from the worker. I fixed that too and this is no longer needed.